src/main/java/org/opentripplanner/framework/io/OtpHttpClient.java

```diff
@@ -485,6 +496,24 @@ private static String sanitizeUri(URI uri) {
     return uri.toString().replace('?' + uri.getQuery(), "");
   }
 
+  private void logResponse(ClassicHttpResponse response) {
+    try {
+      if (
+        log.isTraceEnabled() &&
+        response.getEntity() != null &&
+        response.getEntity().getContent() != null
+      ) {
+        var entity = response.getEntity();
+        String content = new BufferedReader(new InputStreamReader(entity.getContent()))
+          .lines()
+          .collect(Collectors.joining("\n"));
+        log.trace("HTTP request failed with status code {}: \n{}", response.getCode(), content);
```

Comment:
I didn't add a limit for how many characters we log from the response nor validation of the response data format. Do we need those?

Reply:
I would say that if you explicitly set your logging to TRACE then you can expect even very long log messages.